diff --git a/node_modules/@magento/venia-ui/lib/util/makeUrl.js b/node_modules/@magento/venia-ui/lib/util/makeUrl.js
index 3881492..e0f5ea0 100644
--- a/node_modules/@magento/venia-ui/lib/util/makeUrl.js
+++ b/node_modules/@magento/venia-ui/lib/util/makeUrl.js
@@ -64,7 +64,7 @@ const makeOptimizedUrl = (path, { type, ...opts } = {}) => {

     const { origin } = window.location;
     const isAbsolute = absoluteUrl.test(path);
-    const magentoBackendURL = process.env.MAGENTO_BACKEND_URL;
+    const magentoBackendURL = new URL(process.env.MAGENTO_BACKEND_URL);
     let baseURL = new URL(path, mediaBackend);

     // If URL is relative and has a supported type, prepend media base onto path
@@ -75,9 +75,9 @@ const makeOptimizedUrl = (path, { type, ...opts } = {}) => {
         }
     }

-    if (baseURL.href.startsWith(magentoBackendURL) && !useBackendForImgs) {
+    if (baseURL.origin == magentoBackendURL.origin && !useBackendForImgs) {
         // Replace URL base so optimization middleware can handle request
-        baseURL = new URL(baseURL.href.slice(magentoBackendURL.length), origin);
+        baseURL = new URL(baseURL.href.slice(baseURL.origin.length), origin);
     }

     // Append image optimization parameters
