diff --git a/node_modules/@magento/upward-js/lib/resolvers/ProxyResolver.js b/node_modules/@magento/upward-js/lib/resolvers/ProxyResolver.js
index 1b534de..5251362 100644
--- a/node_modules/@magento/upward-js/lib/resolvers/ProxyResolver.js
+++ b/node_modules/@magento/upward-js/lib/resolvers/ProxyResolver.js
@@ -26,9 +26,12 @@ class ProxyResolver extends AbstractResolver {
             this.visitor.upward(definition, 'target'),
             definition.ignoreSSLErrors
                 ? this.visitor.upward(definition, 'ignoreSSLErrors')
-                : false
+                : false,
+            definition.pathRewrite
+                ? this.visitor.upward(definition, 'pathRewrite')
+                : false,
         ];
-        const [targetUrl, ignoreSSLErrors] = await Promise.all(toResolve);
+        const [targetUrl, ignoreSSLErrors, pathRewrite] = await Promise.all(toResolve);
 
         debug('resolved target %o', targetUrl);
         if (typeof targetUrl !== 'string' && !targetUrl.href) {
@@ -48,7 +51,8 @@ class ProxyResolver extends AbstractResolver {
                 cookieDomainRewrite: '',
                 changeOrigin: true,
                 secure: !ignoreSSLErrors,
-                target: targetUrl.toString()
+                target: targetUrl.toString(),
+                pathRewrite: pathRewrite ? JSON.parse(pathRewrite) : null
             };
 
             if (target.username) {
